functions/helpers.bash

```diff
@@ -558,11 +558,148 @@ set_cpu_speed() {
 }
 
 
-## check if InfluxDB is installed
+## Functions to check it a given supported program is installed should be placed below here
+
+## Check if amanda is installed
+##
+##    amanda_is_installed
+##
+amanda_is_installed() {
+  if dpkg -s 'amanda-common' 'amanda-server' 'amanda-client' &> /dev/null; then return 0; fi
+  return 1
+}
+
+## Check if InfluxDB is installed
 ##
-##    iinfluxdb_is_installed
+##    influxdb_is_installed
 ##
 influxdb_is_installed() {
-  if [[ -d /var/lib/influxdb ]]; then return 0; fi
+  if dpkg -s 'influxdb' &> /dev/null && [[ -d /var/lib/influxdb ]]; then return 0; fi
+  return 1
+}
+
+## Check if Grafana is installed
+##
+##    grafana_is_installed
+##
+grafana_is_installed() {
+  if dpkg -s 'grafana' &> /dev/null && [[ -s /etc/grafana/grafana.ini ]]; then return 0; fi
+  return 1
+}
+
+## Check if node is installed
+##
+##    node_is_installed
+##
+node_is_installed() {
+  if [[ -x $(command -v npm) ]] || [[ $(node --version) == "v14"* ]]; then return 0; fi
```

Comment:
is it still correct to test for v14 ? I recall there were issues without but I guess we should at least query for a version equal *or greater* 14

Reply:
Yes it still works, we currently use node 14 as our supposed baseline, I moved it from 12 about a year ago. I don't think we want to do greater than or equal to 14 because there are breaking changes with some of the packages we use between major revisions of the node version we use. It also makes sure that we don't screw someones version up by checking for nom being installed, because if it is we don't even look at the node version.

That whole string was actually taken from the install function to begin with so I think it should be fine.